src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -2562,6 +2580,7 @@ internal struct EventMetadata
             }
             catch (Exception e)
             {
+#if FEATURE_MANAGED_ETW
```

Comment:
This actually should allow ReportOutOfBandMessage (this change clips it out).  As it is, UNIX will not report some errors to the EventListeners, which is bad.   It turns out that changes in the Roxel branch happen to have fixed this (It removes the checks for m_provider=null and instead puts them in the ETW part of ReportOutOfBandMessage().  

However for now, I think you should put the #ifdefs around just the if test.